SharpBucket/V2/EndPoints/EndPoint.cs

```diff
@@ -20,6 +23,12 @@ public EndPoint(SharpBucketV2 sharpBucketV2, string resourcePath)
             _baseUrl = resourcePath;
         }
 
+        public EndPoint(ISharpBucketV2 sharpBucketV2, string resourcePath)
```

Comment:
@MitjaBezensek do we want to add new constructors to avoid any sort of breaking change
Or can we do an assembly-binding level breaking change for a future release where we have only constructors that take the interfaces ?

Since concrete implementation implements the interface, any code that is recompiled against a version that have only constructors that take the interface will be able do to it without seeing the breaking change

But in case of a lib (let name it XBucket) that was compiled against a SharpBuket version that had the concrete type parameter in the constructor, and a product that reference a newer version of SharpBucket and XBucket, it will fail at run time because assembly binding won't be able to find back the exact same constructor

Since we are not already in a V1.0 but in a V0.7 of SharpBucket, I personally found acceptable that kind of small breaking change for a V0.8 release, and you ?

Reply:
I also find it acceptable. It's not a big change, so the users whose code breaks will be able to update quite quickly. Let's just make sure we address the change in the release notes so that they are aware of it an know what to do about it.